### ISSUE TITLE
Extra space in buffers when .setup called?

### ISSUE THREAD

**Issue by richchurcher** [ibhagwan/fzf-lua#832]
### Info

- Operating System: MacOS, Arch Linux
- Shell: zsh
- Terminal: kitty
- `nvim --version`: 0.9.1
- `fzf --version`: 0.4.2

- [ ] The issue is reproducible with `mini.sh`

<details>
<summary>fzf-lua configuration</summary>

```lua

require('fzf-lua').setup()

```
</details>

### Description 

I've noticed something similar to https://github.com/ibhagwan/fzf-lua/issues/814, (extra padding when using `buffers`) which is in fact fixed nicely when using `mini.sh`:
![Screenshot 2023-08-02 at 2 59 27 PM](https://github.com/ibhagwan/fzf-lua/assets/171905/f7797ed9-f084-43f5-8663-4e792fc26d44)
However, with any call to `.setup()` whatsoever, even an empty one, this happens:
![Screenshot 2023-08-02 at 2 52 18 PM](https://github.com/ibhagwan/fzf-lua/assets/171905/81486077-efeb-40d2-b130-a95d2dd59ce9)
Am I perhaps inadvertently overwriting config? Could be user error, but I've tried the most minimal config I can think of, and reproduced the issue.

I'm otherwise an extremely happy plugin user, thanks for all your hard work!

**Comment by ibhagwan** [comment]
>  Am I perhaps inadvertently overwriting config? Could be user error, but I've tried the most minimal config I can think of, and reproduced the issue.


Seems to me like there’s an issue with coloring not being applied (current buffer `%` isn’t red), the ansi escape sequencing adds a lot of invisible characters which are then aligned with `string.format`, when the coloring sequences are missing the spacing will appear much wider.

- Are you calling `setup()` with no arguments?
- `mini.sh` is also using `setup`, does it also reproduce with the mini script?
- Can you call run `:lua require'fzf-lua'.setup_highlights()` and then buffers?
- Can you run `:FzfLua highlights` and post all the highlights starting with `FzfLua…`?

**Comment by richchurcher** [comment]
Yeah I did notice the highlight change, you're right. Lemme see if I can come up with a foolproof repro.

**Comment by ibhagwan** [comment]
> Yeah I did notice the highlight change, you're right. Lemme see if I can come up with a foolproof repro.

My guess is the highlight groups are returning invalid colors.

Try this step:
> Can you run `:FzfLua highlights` and post all the highlights starting with `FzfLua…`?

It would also be helpful to see the the highlight group details from the preview for the `FzfLuaBuf…` groups.

**Comment by richchurcher** [comment]
Yikes. OK, this is weird. There are _no_ `FzfLua` highlight groups if I have any kind of `require("fzf-lua").setup(...` call in my config. But take it out, and they all mysteriously appear (26 groups in total). This is testing with `.setup()` (no args), but the same thing happens with profiles, or with my usual config which is just:
```lua
require "fzf-lua".setup {
  files = {
    fd_opts = '-t f -H -I -E "{.git,tags,dist,__pypackages__}"'
  }
}
```
This definitely does not happen with `mini.sh`.
> Can you call run :lua require'fzf-lua'.setup_highlights() and then buffers?
Yep, this fixes it! Hrm. Incompatibility with another plugin?

**Comment by ibhagwan** [comment]
> Yikes. OK, this is weird. There are no FzfLua highlight groups if I have any kind of require("fzf-lua").setup(... call in my config. But take it out, and they all mysteriously appear (26 groups in total). This is testing with .setup() (no args), but the same thing happens with profiles, or with my usual config which is just:

In order to avoid forcing the user to call `setup`, highlight setup is called twice, once when loading the plug-in:

https://github.com/ibhagwan/fzf-lua/blob/2fa4913c7db0c22e02c003c6f09b7ebb2d0ed367/lua/fzf-lua/init.lua#L106-L109

And **optionally a second time** as part of `setup`:
https://github.com/ibhagwan/fzf-lua/blob/2fa4913c7db0c22e02c003c6f09b7ebb2d0ed367/lua/fzf-lua/init.lua#L150

Now why the call from setup fails in your case is unclear but I would suspect plug-in load ordering:
- What plug-in manager so you use (lazy or packer?)
- Are you lazy loading the colorscheme, fzf-lua or both?

**Comment by richchurcher** [comment]
Using packer, and unless I've misunderstood, I don't _think_ I'm lazy loading either the colorscheme or fzf-lua. But now I'm able to repro the issue with this config (no explicit `.setup` call) which leads me to believe something in my environment is off:
```lua
-- init.lua
require("plugins")
vim.api.nvim_set_keymap("n", "<C-\\>", [[<Cmd>lua require"fzf-lua".buffers()<CR>]], {})

-- lua/plugins.lua
vim.cmd([[
  augroup packer_user_config
    autocmd!
    autocmd BufWritePost plugins.lua source <afile> | PackerSync
  augroup end
]])

return require("packer").startup(function(use)
  use "wbthomason/packer.nvim"

  use { "ibhagwan/fzf-lua", requires = { "nvim-tree/nvim-web-devicons" } }
end)
```
At this stage it doesn't look like an fzf-lua problem, but I appreciate your debugging assistance thus far!

**Comment by ibhagwan** [comment]
> At this stage it doesn't look like an fzf-lua problem, but I appreciate your debugging assistance thus far!

It might be something in your config, but I’d still argue this shouldn’t happen and **might be fixed from fzf-lua**, these lines create the highlights:
https://github.com/ibhagwan/fzf-lua/blob/2fa4913c7db0c22e02c003c6f09b7ebb2d0ed367/lua/fzf-lua/init.lua#L69-L73

It would be interesting to see if these get called and create the highlights, if it fails then I can probably find out why and fix it from fzf-lua.

The other **and more likely cause**, is that something in your config is calling `hi! clear` which removes all highlight groups and that’s why calling `setup_highlights` again solves the issue.

Clearing highlights is usually called at the start of setting up a colorscheme, maybe you can add some prints in your config and see which gets called first, fzf-lua or the colorscheme plug-in (although it technically could be any plug-in).

Although I’m not a big fan of workarounds you could add a call to `setup_highlights` at the end of your config (after all plugins are loaded) which will also solve the issue.

**Comment by richchurcher** [comment]
Yeah, I'll keep looking, if I find anything I'll attempt a PR. It almost feels like something is getting cached, though I've removed everything in the usual places (`.cache/nvim`, `.local/share/nvim`).

**Comment by ibhagwan** [comment]
@richchurcher, any findings why the highlights are getting cleared (or err during creation)?

**Comment by richchurcher** [comment]
I'm sorry for not getting back... I had to unexpectedly job hunt (employer funding crisis, all too common these days!) which has been taking up all available RAM. I'll revisit soon :crossed_fingers:

**Comment by ibhagwan** [comment]
> I'm sorry for not getting back... I had to unexpectedly job hunt (employer funding crisis, all too common these days!) which has been taking up all available RAM. I'll revisit soon 🤞

GL friend!

**Comment by patnr** [comment]
I had the same issue. Using [LazyVim distro](https://github.com/LazyVim/LazyVim)
Correctly specifying the colorscheme fixed it.
Previously I had
```lua
require("config.lazy")
vim.cmd("colo catppuccin")
```

Now I removed the second line. Instead, my `colorscheme.lua` contains:

```
return {
  {
    "LazyVim/LazyVim",
    opts = {
      colorscheme = "catppuccin",
    },
  },
}
```

**Comment by ibhagwan** [comment]
Ty for the the info @patnr!

**Comment by richchurcher** [comment]
Oh, interesting. I'm not a LazyVim user (although see below) so can't comment on that aspect.

I did recently switch from Packer (which is now unmaintained) to Lazy's package manager https://github.com/folke/lazy.nvim and I note the problem has disappeared. I know that's not very helpful in solving the underlying problem, but it does provide something to trial for those who might also be seeing this issue. I also switched colo to https://github.com/edeneast/nightfox.nvim, as another data point.

Happy for this to be closed unless you'd like to use it to track further reports.
